package/src/Luna/Package.hs

```diff
@@ -290,31 +301,61 @@ rename src target = Exception.rethrowFromIO @Path.PathException $ do
     unless_ isValidPkgName . Exception.throw . InvalidName $ convert newName
 
     -- Guaranteed to be in a package by now so default value is nonsensical
-    defaultDir     <- Directory.getCurrentDirectory
-    defaultDirPath <- Path.parseAbsDir defaultDir
+    defaultDir     <- liftIO $ Directory.getCurrentDirectory
+    defaultDirPath <- Exception.rethrowFromIO @Path.PathException
+        $ Path.parseAbsDir defaultDir
     srcPackageRoot <- fromJust defaultDirPath <$> findPackageRoot src
     originalName   <- name srcPackageRoot
 
-    liftIO $ Directory.renameDirectory (Path.fromAbsDir srcPackageRoot) destPath
+    -- Determine things to copy and copy/create them
+    let recursiveListDir dir = do
+            contents <- map (dir `FilePath.combine`)
+                <$> (liftIO $ Directory.listDirectory dir)
+            (files, dirs) <- Utilities.partitionM
+                (liftIO . Directory.doesFileExist) contents
+            (recFiles, recDirs) <- fmap unzip $ mapM recursiveListDir dirs
+            pure (files <> concat recFiles, dirs <> concat recDirs)
+
+    (filesToCopy, dirsToCreate) <- recursiveListDir srcPath
+
+    for_ dirsToCreate $ \dir -> do
+        let relPath = FilePath.makeRelative srcPath dir
+        liftIO . Directory.createDirectoryIfMissing True
+            $ destPath `FilePath.combine` relPath
+
+    for_ filesToCopy $ \file -> do
+        let relPath = FilePath.makeRelative srcPath file
+        liftIO . Directory.copyFile file $ destPath `FilePath.combine` relPath
+
+    Exception.catchAll (\(_ :: SomeException) ->
```

Comment:
I'll try to not discard exceptions, maybe add SomeException as a field to CannotDelete? It could contain valuable information about underlying cause